tools/shared.py

```diff
@@ -1887,8 +1887,29 @@ def llvm_backend_args():
 
     return args
 
+  @staticmethod
+  def link_to_object(linker_inputs, target):
+    # link using lld for the wasm backend with wasm object files,
+    # other otherwise for linking of bitcode we must use our python
+    # code (necessary for asm.js, for wasm bitcode see
+    # https://bugs.llvm.org/show_bug.cgi?id=40654)
+    if Settings.WASM_BACKEND and Settings.WASM_OBJECT_FILES:
+      Building.link_lld(linker_inputs, target, ['--relocatable'])
+    else:
+      Building.link(linker_inputs, target)
```

Comment:
I wonder if this should be a direct call to link_llvm?

Reply:
We need to handle archives etc. here, which `llvm-link` can't do.

Reply:
I guess I was just wondering why Building.link_llvm was factored out here?

Reply:
It just seemed nicer that way, to have a function dedicated to running `llvm-link`. That way `link_llvm` parallels `link_lld`, both invoke the respective linker.